### DISCUSSION TITLE
how do i manually put binaries into mongodb-binaries

### DISCUSSION THREAD

**Discussion by Diluka** [typegoose/mongodb-memory-server#540] [Q&A] [answered]
## Versions

- NodeJS: 0.0.0
- mongodb-memory-server-*: 0.0.0
- mongodb: 0.0.0
- mongoose: 0.0.0 
- system:  macos

package: mongo-memory-server 

## What is your question?

it can't auto download the binaries on my computer. I can download the binaries package by using download app. But how can i put it?

**Comment by hasezoey** [comment, marked as answer]
place the binary where ever needed (probably package manager set), and use the [`SYSTEM_BINARY`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#system_binary) config option in either the [`package.json`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-in-the-packagejson) or as a [environment variable](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-as-environment-variables)

**Reply by Diluka** [reply]
Can I put it into `mongodb-binaries`? As [doc](https://nodkz.github.io/mongodb-memory-server/docs/guides/faq#do-binaries-get-automatically-deleted) says, the cached binaries will keep until manually remove them. If I manually set it, no more configs will need during installing. I install this package globally and indirectly.

**Reply by hasezoey** [reply]
you *could* put it in the MMS cache, though it will not be read by default when not having the exact name that is queried for (platform-distro-arch-version combination)

and yes, this library does not delete binaries

**Reply by Diluka** [reply]
i try to put the package or extracted folder, but it still trying to download a new one with the same name

**Reply by hasezoey** [reply]
as what name did you try to put it in?

if you dont know what to use, you can try to [enable the debug logger](https://nodkz.github.io/mongodb-memory-server/docs/guides/enable-debug-mode) and look what path it tries to look for (search for `DryBinary` logs)

**Reply by Diluka** [reply]
```log
➜  ~ MONGOMS_DEBUG=1 npm i -g mongodb-memory-server

> mongodb-memory-server@7.4.0 postinstall /Users/diluka/.nvm/versions/node/v14.17.3/lib/node_modules/mongodb-memory-server
> node ./postinstall.js

Mongodb-Memory-Server* checking MongoDB binaries
Mongodb-Memory-Server* found binary: "/Users/diluka/.cache/mongodb-binaries/mongod-x64-darwin-4.0.25"
+ mongodb-memory-server@7.4.0
added 72 packages from 35 contributors in 46.116s
```

I see. it just the binary file and log when the download success. and my network is back to normal

**Reply by hasezoey** [reply]
> I see. it just the binary file and log when the download success. and my network is back to normal

actually, the `DEBUG` option, when enabled, should show more than this - what you see here is just the normal output it would give even without the option

**Comment by aderchox** [comment]
I want to thank you @hasezoey (and @Diluka for asking this) and I also want to recommend mentioning this on the README file or at list in the quick start guide this is really confusing that only downloadDir is mentioned.

**Reply by hasezoey** [reply]
yes probably should be, the QSG would also need some rework, just for reference [Here are all config options](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options)
